numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -553,6 +557,13 @@ PyArray_OrderConverter(PyObject *object, NPY_ORDER *val)
     }
     else {
         str = PyBytes_AS_STRING(object);
+        if (strlen(str) != 1) {
+            DEPRECATE("Non length-one string passed in " \
```

Comment:
DEPRECATE needs to be
```
if (DEPRECATE("blabla") < 0) {
    cleanup_code/goto;
    return;
}
```
since it raises an error when the warnings are set accordingly. Also, please add a short test for both deprecations into the `test_deprecations.py` file, that would also find such an issue.